First draft of crewai integrated travel planner

amitabhbanerjee opened the pull request:
Building a travel planner with gradio and crewai integration.

snova-francescar commented on integrations/crewai/travel_planner_crew_sambanova/config/tasks.yaml:
```diff
@@ -0,0 +1,94 @@
+interest_scout_task:
+  description: >
+    Run specifically for the interests of the traveller provided {interests}. As an example if the traveller has interest in museums, plan activities related to museums at the destination.
+    Check if children are accompanying in the trip. {children}
```
Is this children variable a boolean?

amitabhbanerjee replied:
Yes.

snova-francescar commented on integrations/crewai/travel_planner_crew_sambanova/config/tasks.yaml:
```diff
@@ -0,0 +1,94 @@
+    Do a great job so that you can be proud of your work.
+  expected_output: >
+    A detailed itinerary document, the itinerary should include a day-by-day
+    plan with activities, restaurants, and scenic locations. Include the address for each location so that an user can easily go to the address. A day can contain multiple activities. In such a case a recommened mode of transport should be provided to help move from one activity location to another.
```
a user

snova-francescar replied:
recommended

snova-francescar commented on integrations/crewai/travel_planner_crew_sambanova/crew.py:
```diff
@@ -0,0 +1,204 @@
+            config=self.agents_config['personalized_activity_planner'],
+            llm=llm,
+            max_iter=1,
+            tools=[SerperDevTool(), ScrapeWebsiteTool()],  # Example of custom tool, loaded at the beginning of file
```
Could you comment what each tool does?

snova-francescar replied:
Maybe at the beginning of the file, as these tools are used by several agents.

amitabhbanerjee replied:
Okay, documented at the beginning of the file, and added references to the crew documentation of the different tools.

snova-francescar commented on integrations/crewai/travel_planner_crew_sambanova/app.py:
```diff
@@ -0,0 +1,170 @@
+import plotly.graph_objects as go
+from gradio_calendar import Calendar
+
+from .crew import AddressSummaryCrew, TravelCrew
```
Could you switch to absolute instead of relative imports?

snova-francescar replied:
Also below.

snova-francescar reviewed (commented):
This is awesome work! Thank you so much for such a nice feature @amitabhbanerjee 
Only minor comments to address.

snova-jorgep commented:
@amitabhbanerjee can you please move this PR to this repo https://github.com/sambanova/integrations ? now integration examples are maintained there, and integrations folder in AISK will be removed soon

cc @snova-francescar, @rachelbakkke

snova-amitk commented:
@snova-jorgep Lets have someone from the team own this merge and fix. Lets not have @amitabhbanerjee not spend too much time on the process. Thanks!

snova-francescar commented on base-requirements.txt:
```diff
@@ -58,6 +58,9 @@ nest_asyncio==1.6.0
 httpx==0.27.0
 backoff==2.2.1
 streamlit-javascript==0.1.5
+gradio==5.1.0
```
@snova-kwasia Do we need to add the gradio dependencies to the `base-requirements.txt`?

snova-francescar replied:
Let's leave the dependency in `base-requirements.txt` for now.

amitabhbanerjee commented on integrations/crewai/travel_planner_crew_sambanova/README.md:
````diff
@@ -0,0 +1,56 @@
+git clone https://github.com/sambanova/ai-starter-kit.git
+```
+
+Navigate to the *travel-planner-crew-sambanova* folder. Then run pip install -r requirements.txt:
````
I re-wrote the entire part with instructions on how to run using venv.

amitabhbanerjee replied:
Actually, I hadn't tested the sequence of commands with uv. Just rewrote this section on the lines of using pip and venv.

amitabhbanerjee commented:
> @snova-jorgep Lets have someone from the team own this merge and fix. Lets not have @amitabhbanerjee not spend too much time on the process. Thanks!

I will address comments on this PR and commit to this PR. Hopefully someone can help move it to the correct git repo.

amitabhbanerjee commented on integrations/crewai/travel_planner_crew_sambanova/README.md:
```diff
@@ -0,0 +1,56 @@
+- You can get your Serper API Key [here](https://serper.dev/)
+
+### Running
+- Start the Gradio App by running python3 app.py
```
No need to cd... Just use default folder.

amitabhbanerjee commented on integrations/crewai/travel_planner_crew_sambanova/app.py:
```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
```
Deleted.

amitabhbanerjee commented on integrations/crewai/travel_planner_crew_sambanova/README.md:
```diff
@@ -0,0 +1,56 @@
+
+
+### API Key Setup 
+**Before you begin Add your `SAMBANOVA_API_KEY` and `SERPER_API_KEY` into the `.env` file in the main starter kit repo folder. Alternatively provide them as environment variables.**
```
I meant alternatively, provide while running the app. Clarified.

amitabhbanerjee commented:
@snova-francescar I addresses all your feedback (Many thanks for that), and the tests are now passing. Let me know how we want to handle the remaining unresolved comments
